clang/lib/Parse/ParseTemplate.cpp

```diff
@@ -787,6 +787,20 @@ NamedDecl *Parser::ParseTemplateTemplateParameter(unsigned Depth,
                                                   unsigned Position) {
   assert(Tok.is(tok::kw_template) && "Expected 'template' keyword");
 
+  if (Token ahead = GetLookAheadToken(1); ahead.isNot(tok::less)) {
+    // Maybe they intended `typename` instead of `template` (given thats more
+    // common) Error early, to add a fixit hint
```

Comment:
```suggestion
    // `template` may have been a typo for `typename`, given that the
    // latter is more common.
```
nit: the comment could be cleaned up a bit, maybe like this

Reply:
Ah yes that comment is more consistent with other wording I've seen, thanks for the suggestion